### ISSUE TITLE
Localization of WGDashboard

### ISSUE THREAD

**Issue by donaldzou** [donaldzou/WGDashboard#397]
Hi all, I finally finished adding the support of localization, and currently WGDashboard support the following languages:
- English
- Chinese (Simplified)
- Chinese (Traditional)

I'm hoping if any of you are willing to contribute, you can follow the instructions below:

1. Download the `v4.1` branch
2. Copy one of the existing locale `.json` file and rename them using the locale code in this list: https://www.science.co.il/language/Locale-codes.php
3. The key is the matching what the dashboard have, so don't change it, and the value is the translation.
4. Start translate ;)
5. Add your translation to `src/static/locale/active_languages.json` by following the same format

> You might notice there are some Regex syntax in the existing locale files. Basically, the dashboard will match the key to the texts in the UI, and replace the value using regex. So you can adjust the position of matching groups to better fit your language.
> For example: `You can add up to (.*) peers` have one matching group which is matching a number between **to** and **peers**, and you can use `$1` in the value to represent the number.

Please comment below if you have any questions 😄

**Comment by donaldzou** [comment]
> I can provide Dutch if you'd like, but how would I do that?

Hi @DaanSelen, I provided an instruction above :) Is fairly easy as I already have an example in the `v4.1-dev` branch. Basically the file looks like this:

```json
{
	"Welcome to": "歡迎來到",
	"Username": "用戶名",
	"Password": "密碼",
	"OTP from your authenticator": "您多重身份驗證器的一次性驗證碼",
	"Sign In": "登錄"
...
}

```
Where the key is what you translating from, and the value is what you are translating to. The example provided is Chinese (Traditional)

**Comment by NOXCIS** [comment]
@donaldzou I can do the Spanish bit.

**Comment by orangeferdi** [comment]
@donaldzou I will support with German! :)

**Comment by donaldzou** [comment]
Hi @3vis97 ! Thank you for providing the Italian translation. Do you mind creating a PR to my `v4.1-dev` branch so I can merge your work to mine?

**Comment by 3vis97** [comment]
> Hi @3vis97 ! Thank you for providing the Italian translation. Do you mind creating a PR to my `v4.1-dev` branch so I can merge your work to mine?

Hi @donaldzou!
Yes i will do, but before I create the PR, I'd like to test the translation in a live instance to ensure the female, male and plural forms are correct and the translation sounds natural in the web interface.

Yes i will do, but before I create the PR, I'd like to test the translation in a live instance to ensure the female, male and plural forms are correct and the translation sounds natural in the web interface.

I only need to find the time to do it. I hope to do it in the next few days 👍🏻

**Comment by donaldzou** [comment]
> > Hi @3vis97 ! Thank you for providing the Italian translation. Do you mind creating a PR to my `v4.1-dev` branch so I can merge your work to mine?
> 
> Hi @donaldzou! Yes i will do, but before I create the PR, I'd like to test the translation in a live instance to ensure the female, male and plural forms are correct and the translation sounds natural in the web interface.
> 
> I only need to find the time to do it. I hope to do it in the next few days 👍🏻

For sure! Please take your time and can't thank you enough for providing this ;)

**Comment by 3vis97** [comment]
> > > Hi @3vis97 ! Thank you for providing the Italian translation. Do you mind creating a PR to my `v4.1-dev` branch so I can merge your work to mine?
> > 
> > 
> > Hi @donaldzou! Yes i will do, but before I create the PR, I'd like to test the translation in a live instance to ensure the female, male and plural forms are correct and the translation sounds natural in the web interface.
> > I only need to find the time to do it. I hope to do it in the next few days 👍🏻
> 
> For sure! Please take your time and can't thank you enough for providing this ;)

Hi @donaldzou, Here's the the PR https://github.com/donaldzou/WGDashboard/pull/406 !

**Comment by donaldzou** [comment]
> > @donaldzou I can do the Spanish & Portuguese bit.
> 
> 
> 
> I can do Brazilian Portuguese

@velrino For sure! Thanks is advanced.

**Comment by jursed** [comment]
I added a Czech translation, see PR https://github.com/donaldzou/WGDashboard/pull/452. :slightly_smiling_face:

**Comment by DaanSelen** [comment]
How can I verify the made translations? Can I already toggle it somehow - somewhere in a configuration file.

**Comment by donaldzou** [comment]
Hi Daan,


Yes if you use the `v4.1-dev` branch, you can toggle it in “Settings” => “WGDashboard Settings” => “Appearance” => “Language” ;)


Best,

Donald



On Nov 7, 2024, at 5:57 PM, dselen ***@***.***> wrote:



How can I verify the made translations? Can I already toggle it somehow - somewhere in a configuration file.

**Comment by oskarax** [comment]
I can do Swedish if its needed?

**Comment by DaanSelen** [comment]
Yes! Of course, do you know how to contribute?

**Comment by oskarax** [comment]
> Yes! Of course, do you know how to contribute?

Not really. There is some instruction at the top and I can try and see if I get it.

**Comment by DaanSelen** [comment]
Sure, let me know if you run into problems. Fastest way is Discord or matrix

**Comment by oskarax** [comment]
> 6\. After you're done, create an PR to the main :)

So its done now. My first time so please be gentle. I am learning :)

**Comment by oskarax** [comment]
> Sure, let me know if you run into problems. Fastest way is Discord or matrix

So its done now. My first so please be gentle. I am learning :)

**Comment by DaanSelen** [comment]
Of course!

**Comment by oskarax** [comment]
> Of course!

i have uploaded the new json file to my test system and it seems to work as it should.

**Comment by XigmaDev** [comment]
I can do Persian
do you need it ?

**Comment by DaanSelen** [comment]
Yeah! Make a PR

**Comment by sobhydo** [comment]
I'm doin the Arabic translation, is it needed ?

**Comment by DaanSelen** [comment]
> I'm doing the Arabic translation, is it needed ?

Always!

**Comment by karorogunso** [comment]
Hey. I translate the Thai language. I made the PR.

**Comment by ahmadmuqri0** [comment]
I can do Malay translation, need any?

**Comment by DaanSelen** [comment]
> I can do Malay translation, need any?

You are very welcome! Look at the instructions above and submit a PR. You can ask for help in Discord (or Matrix) or here.

**Comment by reykim7854** [comment]
Hi @donaldzou Thank you for starting this project! really helped me out a lot in managing my WG peers, I'd like to contribute a bit by adding Bahasa Indonesia to the language localization, please help to review my PR #722, cheers!

**Comment by karorogunso** [comment]
I will update the thai translation soon. please wait.

**Comment by DaanSelen** [comment]
> I will update the thai translation soon. please wait.

Thanks!

**Comment by mghextreme** [comment]
@donaldzou Added a PR for PT-BR translation: #817 

@NOXCIS @velrino You mentioned helping with Portuguese. Would be great to have you review my PR.

**Comment by DukeSoft55** [comment]
It seems to me that in one of the latest versions (v4.3.0.1) the drop-down list button for selecting a language does not work.

<img width="1781" height="279" alt="Image" src="https://github.com/user-attachments/assets/e536d83a-9e53-4be5-92f4-876d287a1d4e" />

The second problem is that for a long time no one has translated the page with translation into Russian (Link [src/static/locales/ru-RU.json](https://github.com/WGDashboard/WGDashboard/blob/main/src/static/locales/ru-RU.json)) and some of the pages remain untranslated..

**Comment by DaanSelen** [comment]
> It seems to me that in one of the latest versions (v4.3.0.1) the drop-down list button for selecting a language does not work.
> 
> The second problem is that for a long time no one has translated the page with translation into Russian (Link [src/static/locales/ru-RU.json](https://github.com/WGDashboard/WGDashboard/blob/main/src/static/locales/ru-RU.json)) and some of the pages remain untranslated..

I tested this version in Docker and for me it seems to work fine, the language switching works to different languages.
However, not all languages are evenly translated. And we fully rely on community input for those translations.

So if you want you can help us by translating missing lines into whatever language you speak. I should do so as well for Dutch.
